src/main.rs

```diff
@@ -631,23 +626,12 @@ fn truncate(s: &str, n: usize) -> String {
 
     if s.len() <= n { return String::from(s) }
 
-    let mut truncated = String::new();
-    for grapheme in UnicodeSegmentation::graphemes(s, true).take(n - tail.len()) {
-        truncated.push_str(grapheme)
-    }
-    truncated.push_str(tail);
-    truncated
+    UnicodeSegmentation::graphemes(s, true).take(n - tail.len()).collect::<String>() + tail
```

Comment:
Small note on the `+`: it has to be `String + &str`. If you have two `String`s, you can do `String + &String` and `&String` coerces to `&str` thanks to `Deref`.